src/commonpp/core/Gelf.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * File: src/commonpp/core/LoggingInterface.cpp
+ * Part of commonpp.
+ *
+ * Distributed under the 2-clause BSD licence (See LICENCE.TXT file at the
+ * project root).
+ *
+ * Copyright (c) 2018 Thomas Sanchez.  All rights reserved.
+ *
+ */
+
+#include "commonpp/core/LoggingInterface.hpp"
+
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+
+#include <boost/container/flat_map.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/log/attributes/function.hpp>
+#include <boost/log/core.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/expressions/attr.hpp>
+#include <boost/log/expressions/formatters/date_time.hpp>
+#include <boost/log/expressions/formatters/named_scope.hpp>
+#include <boost/log/sinks/sync_frontend.hpp>
+#include <boost/log/sinks/syslog_backend.hpp>
+#include <boost/log/sinks/text_multifile_backend.hpp>
+#include <boost/log/sources/basic_logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/sources/severity_channel_logger.hpp>
+#include <boost/log/sources/severity_logger.hpp>
+#include <boost/log/support/date_time.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/log/utility/setup/common_attributes.hpp>
+#include <boost/log/utility/setup/console.hpp>
+#include <boost/log/utility/setup/file.hpp>
+#include <boost/log/utility/value_ref.hpp>
+
+#include <boost/phoenix/bind.hpp>
+
+#include "commonpp/core/Utils.hpp"
+#include "commonpp/core/string/json_escape.hpp"
+#include "commonpp/thread/Thread.hpp"
+
+#include <boost/asio.hpp>
+#include <boost/asio/ip/udp.hpp>
+namespace logging = boost::log;
+namespace sinks = logging::sinks;
+namespace keywords = logging::keywords;
+namespace expr = logging::expressions;
+namespace attrs = logging::attributes;
+namespace src = logging::sources;
+
+namespace
+{
+
+size_t to_syslog_level(commonpp::LoggingLevel level)
```

Comment:
this one could be noexcept

Reply:
yep, still fighting with boost.log to extract the severity in the formatter.